### PR TITLE
storage, test: unlock transaction in the retry loop

### PR DESCRIPTION
**Pull request by mitake**
Reported by @AkihiroSuda . Could you share how to reproduce the problem, @AkihiroSuda ?

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@mitake, thanks for opening this ticket.

I reproduced the bug by just running `test` script located in the root dir of the etcd repository.
Stack trace: https://gist.github.com/AkihiroSuda/df2a21c2286d8ef2b4f7

Unfortunately, even after applied the patch eb7177f, I hit another problem: https://gist.github.com/AkihiroSuda/abb2db5c8375e863fd7b
(This is not a degration caused by eb7177f, testcase differs.)

File: storage/kvstore_test.go [hunk 1]
```diff
@@ -656,6 +656,7 @@ func TestRestoreContinueUnfinishedCompaction(t *testing.T) {
 		ks, _ := tx.UnsafeRange(keyBucketName, revbytes, nil, 0)
 		if len(ks) != 0 {
 			time.Sleep(100 * time.Millisecond)
+			tx.Unlock()
```
Comment (jonboulle) [inline storage/kvstore_test.go]: Rather just unlock on line 657 immediately after use, then you don't need both.

**Comment by mitake** [comment]
@jonboulle thanks for your comment. I updated the place of tx.Unlock(), PTAL

**Comment by xiang90** [comment]
LGTM. Do we want to wait for @AkihiroSuda's reply?

**Comment by xiang90** [comment]
@AkihiroSuda Actually can you please open another issue for tracking the other issue?

**Comment by AkihiroSuda** [comment]
@xiang90, sorry, I opened #4039

**Pull request merged**